cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -431,6 +431,57 @@ func (s *Server) repoDetail(ctx context.Context, repoRef *corev1.PackageReposito
 	}, nil
 }
 
+func (s *Server) repoSummaries(ctx context.Context, namespace string) ([]*corev1.PackageRepositorySummary, error) {
+	summaries := []*corev1.PackageRepositorySummary{}
+	var repos []sourcev1.HelmRepository
+	var err error
+	if namespace == apiv1.NamespaceAll {
+		if repos, err = s.listReposInAllNamespaces(ctx); err != nil {
+			return nil, err
+		}
+	} else {
+		// here I think the right semantics are different than that of availablePackageSummaries
+		// namely, if a specific namespace is specified we need to list repos in that namespace
```

Comment:
Yes. So for example, even if there are repos in Carvel's `kapp-controller-packaging-global` namespace, we wouldn't want to be returning those when asking for the repos in the "default" namespace, even though for available packages we would. Agree.